src/components/graph/adGroups.tsx

```diff
@@ -0,0 +1,128 @@
+import { Tooltip, Typography } from '@equinor/eds-core-react';
+import * as PropTypes from 'prop-types';
+import AsyncSelect from 'react-select/async';
+import { SimpleAsyncResource } from '../async-resource/simple-async-resource';
+import { useCallback, useEffect, useState } from 'react';
+import { useAuthentication } from './authentication';
+import { adGroupModel } from './adGroupModel';
+import { getGroup, getGroups } from './graphService';
+import { RequestState } from '../../state/state-utils/request-states';
+import { AsyncState } from '../../effects/effect-types';
+import { debounce } from 'lodash';
+
+export interface ADGroupsProps {
+  handleAdGroupsChange: (event: adGroupModel[]) => void;
+  adGroups: string;
+  isDisabled: boolean;
+  adModeAuto: boolean;
```

Comment:
mark optional values optional
```suggestion
  adGroups?: string;
  isDisabled?: boolean;
  adModeAuto?: boolean;
```